### ISSUE TITLE
In POSIX sh, >& is undefined when using variables representing file descriptors

### ISSUE THREAD

**Issue by Earnestly** [koalaman/shellcheck#1515]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2039
- My shellcheck version (`shellcheck --version` or "online"): 0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
fd=3
exec 3>&1
printf 'test\n' >&"$fd"

```

#### Here's what shellcheck currently says:

```
In t.sh line 4:
printf 'test\n' >&"$fd"
                ^-----^ SC2039: In POSIX sh, >& is undefined.

For more information:
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, >& is undefined.
```

#### Here's what I wanted or expected to see:

Nothing.
